### ISSUE TITLE
Ansible Dry-run KO because of non idempotent-task

### ISSUE THREAD

**Issue by nemoclaranet** [claranet/ansible-role-mariadb#2]
Ansible Dry-run KO because of non idempotent-task

## Expected Behavior

After I applied the role with success on my server, I tried to launch a dry-run to check if all is OK)
-> here the expected behavior is a success of a dry-run with eventually some changes.

## Current Behavior

But the dry-run job is KO with the following error :

```
TASK [claranet.mariadb : Get list of hosts for the anonymous user.] ************
skipping: [cetelem.cmonprojet.hostingpp-01]
TASK [claranet.mariadb : Remove anonymous MariaDB users.] **********************
fatal: [cetelem.cmonprojet.hostingpp-01]: FAILED! => {}
MSG:
'dict object' has no attribute 'query_result'. 'dict object' has no attribute 'query_result'
PLAY RECAP *********************************************************************
cetelem.cmonprojet.hostingpp-01 : ok=28   changed=3    unreachable=0    failed=1    skipped=18   rescued=0    ignored=0   
Cleaning up project directory and file based variables 00:01
ERROR: Job failed: exit code 1
```

Because the previous task is skipped in dry-run mod, this task hasn't the register var "query_result" and fails.

## Possible Solution

Maybe the task "claranet.mariadb : Remove anonymous MariaDB users." need to be remove when launched in dry-run mode ?

## Steps to Reproduce (for bugs)

1. Apply the role on your server.
2. Launch a dry-run -> KO

## Your Environment

* Operating System : Debian 11

Thank you,
Regards,

**Comment by nemoclaranet** [comment]
![Capture d'écran 2023-06-26 094526](https://github.com/claranet/ansible-role-mariadb/assets/114510243/37900ede-df33-4511-a637-ae1c336b21c8)

**Comment by davidassigbi** [comment]
Hello @nemoclaranet

I have been able to reproduce the error and have written a fix to skip the 'guilty' tasks when running the role in check mode.

These fixes are part of the next release (out within the next hour).

**Comment by davidassigbi** [comment]
The new version 1.0.1 is released...

Pull the newer version from galaxy and test to confirm the dry-run is working as expected.

**Comment by nemoclaranet** [comment]
Hello,

Thank you for the fix, the dry-run is now OK for me with the 1.0.1 version.

I close this issue.

Regards,
